src/main/java/no/ssb/subsetsservice/SubsetsController.java

```diff
@@ -62,6 +69,175 @@ public SubsetsController(){
         return getFrom(LDS_SUBSET_API, "/"+id);
     }
 
+    @GetMapping("/v1/subsets/{id}/versions")
+    public ResponseEntity<String> getVersions(@PathVariable("id") String id) {
+        return getFrom(LDS_SUBSET_API, "/"+id+"?timeline");
+    }
+
+    /**
+     * Get a list of versions of a subset that starts with {versions}.
+     * So if {versions} is "1.0", then 1.0.0, 1.0.1, 1.0.2 etc will be returned.
+     * If {versions} is "1.0.1", then a list with a single item (1.0.1) will be returned
+     * @param id
+     * @param version
+     * @return
+     */
+    @GetMapping("/v1/subsets/{id}/versions/{version}")
+    public ResponseEntity<JsonNode> getVersion(@PathVariable("id") String id, @PathVariable("version") String version) {
+        ResponseEntity<String> ldsRE = getFrom(LDS_SUBSET_API, "/"+id+"?timeline");
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            JsonNode responseBodyJSON = mapper.readTree(ldsRE.getBody());
+            if (responseBodyJSON != null){
+                if (responseBodyJSON.isArray()) {
+                    ArrayNode responseBodyArrayNode = (ArrayNode) responseBodyJSON;
+                    ArrayNode returnVersionsArrayNode = mapper.createArrayNode();
+                    for (int i = 0; i < responseBodyArrayNode.size(); i++) {
+                        JsonNode arrayEntry = responseBodyArrayNode.get(i).get("document");
+                        String subsetVersion = arrayEntry.get("version").asText();
+                        if (subsetVersion.startsWith(version)){
+                            returnVersionsArrayNode.add(arrayEntry);
+                        }
+                    }
+                    return new ResponseEntity<>(returnVersionsArrayNode, HttpStatus.OK);
```

Comment:
It is not enough.
FIXME: The service should updated metadata fields for every version to be the latest. 
TODO: Each version should include HTTP-reference the same way code list versions from the Klass API do.
There is probably more stuff to be done before sending the response. We have to define requirements as tasks.

Reply:
by HTTP-reference, do you mean this part of the list version?
`"_links" : {
    "self" : {
      "href" : "http://data.ssb.no/api/klass/v1/versions/1"
    }
  }`

Reply:
What do you mean specifically when you say "The service should updated metadata fields for every version to be the latest"?

Reply:
### Imagine history of versions to a subset with id 1:
{id: 1, **v1**, name: A}, 
{id: 1, **v2**, name: A}, 
{id: 1, **v2.0.1**, name: **B**}, 
{id: 1, **v3**, name: B}. 

### The subsets service should return:
- subsets/1/versions/1 -> {id: 1, v1, name: B}
- subsets/1/versions/2 -> {id: 1, v2, name: B}
- subsets/1/versions/3 -> {id: 1, v3, name: B}
-subset/1/versions/ -> [
  {id: 1, v1, name: B}, 
  {id: 1, v2, name: B}, 
  {id: 1, v3, name: B}
]

### Summary
There are some fields in a subset scheme, which are version-independent and "overwrite" the previous value for all versions, including already existent ones. Those are `name`, `validFrom`, `validUntil`, `CreatedBy`, `administrativeDetails.ANNOTATION.values`.